codalab/apps/chahub/tasks.py

```diff
@@ -0,0 +1,221 @@
+    CompetitionSubmission
+from apps.web.utils import inheritors
+
+logger = logging.getLogger(__name__)
+
+
+def _send(endpoint, data):
+    url = "{}{}".format(settings.CHAHUB_API_URL, endpoint)
+    headers = {
+        'Content-type': 'application/json',
+        'X-CHAHUB-API-KEY': settings.CHAHUB_API_KEY,
+    }
+    logger.info("ChaHub :: Sending to ChaHub ({}) the following data: \n{}".format(url, data))
+    return requests.post(url=url, data=json.dumps(data), headers=headers)
+
+
+def get_obj(app_label, model_name, pk, include_deleted=False):
+    Model = models.get_model(app_label, model_name)
+
+    try:
+        if include_deleted:
+            obj = Model.objects.all_objects().get(pk=pk)
+        else:
+            obj = Model.objects.get(pk=pk)
+    except Model.DoesNotExist:
+        raise ChahubException("Could not find {} with pk: {}".format(app_label, pk))
+    return obj
+
+
+@task(queue='site-worker')
+def send_to_chahub(app_label, model_name, pk, data, data_hash):
+    """
+    Does a post request to the specified API endpoint on chahub with the inputted data.
+    """
+    if not settings.CHAHUB_API_URL:
+        raise ChahubException("CHAHUB_API_URL env var required to send to Chahub")
+    if not settings.CHAHUB_API_KEY:
+        raise ChahubException("No ChaHub API Key provided")
+
+    obj = get_obj(app_label, model_name, pk)
+
+    try:
+        resp = _send(obj.get_chahub_endpoint(), data)
+    except requests.exceptions.RequestException:
+        resp = None
+
+    if resp and resp.status_code in (200, 201):
+        logger.info("ChaHub :: Received response {} {}".format(resp.status_code, resp.content))
+        obj.chahub_timestamp = timezone.now()
+        obj.chahub_data_hash = data_hash
+        obj.chahub_needs_retry = False
+    else:
+        status = getattr(resp, 'status_code', 'N/A')
+        body = getattr(resp, 'content', 'N/A')
+        logger.info("ChaHub :: Error sending to chahub, status={}, body={}".format(status, body))
+        obj.chahub_needs_retry = True
+    obj.save(send=False)
+
+
+@task(queue='site-worker')
+def delete_from_chahub(app_label, model_name, pk):
+    if not settings.CHAHUB_API_URL:
+        raise ChahubException("CHAHUB_API_URL env var required to send to Chahub")
+    if not settings.CHAHUB_API_KEY:
+        raise ChahubException("No ChaHub API Key provided")
+
+    obj = get_obj(app_label, model_name, pk, include_deleted=True)
+
+    url = "{}{}{}/".format(settings.CHAHUB_API_URL, obj.get_chahub_endpoint(), pk)
+    logger.info("ChaHub :: Sending to ChaHub ({}) delete message".format(url))
+
+    headers = {'X-CHAHUB-API-KEY': settings.CHAHUB_API_KEY}
+
+    try:
+        resp = requests.delete(url=url, headers=headers)
+    except requests.exceptions.RequestException:
+        resp = None
+
+    if resp and resp.status_code == 204:
+        logger.info("ChaHub :: Received response {} {}".format(resp.status_code, resp.content))
+        obj.delete()
```

Comment:
This is where we wanna pass `real_delete`?

Reply:
added